CameraXBasic/app/src/main/AndroidManifest.xml

```diff
@@ -30,8 +30,12 @@
     <uses-permission android:name="android.permission.CAMERA" />
 
     <application
+        android:icon="@mipmap/ic_launcher"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:name=".CameraXBasic"
```

Comment:
Please don't hard-code strings in the manifest. Remove this line, since it's not necessary.

Reply:
It is required, else there won't be any `CameraProvider` which `CameraXBasic` implements.

Reply:
Then use a string resource, perhaps the app name

Reply:
This is not `${applicationId}` but the relative name of the `Application` class, which is coincidentally (intentionally) the same as the name of the project. Without the leading `.`, it won't resolve.

Reply:
One my side, this rather raises the question: What if there is no camera2? Because it always returns `Camera2Config.defaultConfig()` - which might not be applicable on elder devices.